### PR TITLE
fix: support new 'ECS' deployment type rather than relying on a null value

### LINKED ISSUE #384: Job Failing with "Error: Unsupported deployment controller: ECS"
### Description

Recent jobs (as of around 28 Sep 03:30 UTC) have started failing with the error:

```
Error: Unsupported deployment controller: ECS
```

The specific job in the workflow that's failing is:

```yaml
      - name: Deploy Amazon ECS backend service task definition
        uses: aws-actions/amazon-ecs-deploy-task-definition@v1
        with:
          task-definition: ${{ steps.task-def.outputs.task-definition }}
          service: ${{ env.ECS_SERVICE }}
          cluster: ${{ env.ECS_CLUSTER }}
          wait-for-service-stability: true
          wait-for-minutes: 8 
```

Previous jobs in the workflow (incl. Checkout, ECR Login, Building & pushing image to ECR, Filling in Task definition) are all passing.

Any idea what the problem could be?

#### Additional Notes:
- Deployment directly from the Amazon ECS Web UI is working 
- There have been no changes to workflows since ~25th Sep, and there were successful deployments then.
- Attached is a redacted version of the task definition 

screenshot-20220928-042359

### PR DESCRIPTION
**Pull request by bradens**
Fixes #384 

*Description of changes:*

The service response was returning 'ECS' instead of null in most cases when running. Instead of failing to deploy, we support the "ECS" deployment type instead of only relying on the descriptor to be null. 

Also, when fixing this, i noticed that some of the dependabot prs broke packaging.  I think this might be because the dependencies/npm configuration are very out of date. I've reverted back to the point where packaging was possible.  

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by muya — APPROVED** [review]

**Comment by duksis** [comment]
Would be thankful if this can get released

**Review by markoorn — APPROVED** [review]

File: dist/index.js [hunk 1]
```diff
@@ -556,10 +556,10 @@ async function run() {
         throw new Error(`Service is ${serviceResponse.status}`);
       }
 
-      if (!serviceResponse.deploymentController) {
+      if (!serviceResponse.deploymentController || serviceResponse.deploymentController.type === "ECS") {
```
Comment (vlahanas) [inline dist/index.js:559]: Single quotes for consistency:

```
if (!serviceResponse.deploymentController || serviceResponse.deploymentController.type === 'ECS')
```

File: index.js [hunk 2]
```diff
@@ -308,10 +308,10 @@ async function run() {
         throw new Error(`Service is ${serviceResponse.status}`);
       }
 
-      if (!serviceResponse.deploymentController) {
+      if (!serviceResponse.deploymentController || serviceResponse.deploymentController.type === "ECS") {
```
Comment (vlahanas) [inline index.js:311]: Single quotes for consistency:

```
if (!serviceResponse.deploymentController || serviceResponse.deploymentController.type === 'ECS')
```

**Comment by mariokoehler** [comment]
there's certainly an argument that can be made about being "consistent". there's also an argument that could be made about pragmatism if all over Github thousands of pipeline are broken and a double quote instead of a single quote is the reason why this isn't fixed by now...just sayin'

**Comment by vlahanas** [comment]
@mariokoehler I don't think my comments prevent the PR from merging, I am not even a contributor on these repos. However, for the broken pipelines there is [a workaround](https://github.com/aws-actions/amazon-ecs-deploy-task-definition/issues/384#issuecomment-1263351938)

**Review by vlahanas — APPROVED** [review]

**Comment by mariokoehler** [comment]
> @mariokoehler I don't think my comments prevents this PR from merging, I am not even a contributor on these repos. It just needs an approval by a maintainer.
> 
> However, for the broken pipelines there is [a workaround](https://github.com/aws-actions/amazon-ecs-deploy-task-definition/issues/384#issuecomment-1263351938)



> @mariokoehler I don't think my comments prevents this PR from merging, I am not even a contributor on these repos. It just needs an approval by a maintainer.
> 
> However, for the broken pipelines there is [a workaround](https://github.com/aws-actions/amazon-ecs-deploy-task-definition/issues/384#issuecomment-1263351938)

in that case, my apologies for the misunderstanding and thanks for the workaround link!

**Comment by vlahanas** [comment]
No worries, to be honest I added the comments just to make sure that the PR will not be delayed once a maintainer jumps in.

**Comment by vlahanas** [comment]
> > @mariokoehler I don't think my comments prevents this PR from merging, I am not even a contributor on these repos. It just needs an approval by a maintainer.
> > However, for the broken pipelines there is [a workaround](https://github.com/aws-actions/amazon-ecs-deploy-task-definition/issues/384#issuecomment-1263351938)
> 
> > @mariokoehler I don't think my comments prevents this PR from merging, I am not even a contributor on these repos. It just needs an approval by a maintainer.
> > However, for the broken pipelines there is [a workaround](https://github.com/aws-actions/amazon-ecs-deploy-task-definition/issues/384#issuecomment-1263351938)
> 
> in that case, my apologies for the misunderstanding and thanks for the workaround link!

No worries, to be honest I added the comments just to make sure that the PR will not be delayed once a maintainer jumps in.

**Review by efekarakus — APPROVED** [review]

**Comment by vpontis** [comment]
@efekarakus just a heads up I think you need to run GitHub checks again

**Comment by efekarakus** [comment]
Hi folks, @dannyrandall merged the fix here: https://github.com/aws-actions/amazon-ecs-deploy-task-definition/pull/387. Apologies for the inconvenience 🙇 !

We couldn't add additional commits to this original PR:
```
git push company-z master                                                                                                                                                                                       f65129e 
ERROR: Permission to company-z/amazon-ecs-deploy-task-definition.git denied to dannyrandall.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exist
```
Maybe maintainer edits are not enabled? 💭  We're going to take a look at releasing the action next.

**Comment by bradens** [comment]
Thanks everyone for the comments, I had to go afk for a while immediately after putting up this pr, but I'm glad you got the fix onto main and people can deploy again 😅

**Pull request closed without merging**